src/Features/Core/Portable/EditAndContinue/IActiveStatementProvider.cs

```diff
@@ -0,0 +1,21 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Immutable;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.CodeAnalysis.EditAndContinue
+{
+    /// <summary>
+    /// Provides current active statements.
+    /// </summary>
+    internal interface IActiveStatementProvider
+    {
+        /// <summary>
+        /// 
+        /// Shall only be called while in debug mode.
+        /// </summary>
+        Task<ImmutableArray<ActiveStatementDebugInfo>> GetActiveStatementsAsync(CancellationToken cancellationToken);
```

Comment:
>GetActiveStatementsAsync [](start = 55, length = 24)

maybe rename to GetActiveStatementsDebugInfoAsync.

Reply:
I don't think that's necessary.

---
In reply to: [162479002](https://github.com/dotnet/roslyn/pull/24316#discussion_r162479002) [](ancestors = 162479002)